powerline/segments/common/bat.py

```diff
@@ -95,6 +95,14 @@ def _get_capacity(pl):
 	else:
 		pl.debug('Not using pmset: executable not found')
 
+	if sys.platform.startswith('cygwin'):
+		BATTERY_PERCENT_RE = re.compile('[0-9]+')
+		def _get_capacity(pl):
+			battery_summary = run_cmd(pl, ['WMIC', 'Path','Win32_Battery', 'GET', 'EstimatedChargeRemaining'])
+			battery_percent = BATTERY_PERCENT_RE.search(battery_summary).group(0)
```

Comment:
I am wondering what this line will return if there is *no* battery. It must return `None`, **not** raise an `AttributeError` (if `re.search` have not found anything it will return `None` and `None` has no attribute `group`). I also do not know what `WMIC` will do in this case: if it exits with non-zero status (I guess unlikely) you will get exception earlier, but you should not catch it unless it *does* exit with non-zero code without battery.